pkg/aws/elb.go

```diff
@@ -52,39 +69,74 @@ func (c *Client) getCanonicalZoneIDs(endpoints []*pkg.Endpoint) (map[string]stri
 }
 
 func (c *Client) getELBs(session *session.Session) ([]*LoadBalancer, error) {
+	result := make([]*LoadBalancer, 0)
+
 	client := elb.New(session)
 
-	resp, err := client.DescribeLoadBalancers(nil)
-	if err != nil {
-		return nil, err
+	params := &elb.DescribeLoadBalancersInput{
+		Marker: nil,
 	}
 
-	loadBalancers := resp.LoadBalancerDescriptions
-	result := make([]*LoadBalancer, len(loadBalancers))
-	for i, loadbalancer := range loadBalancers {
-		result[i] = &LoadBalancer{
-			DNSName:         aws.StringValue(loadbalancer.DNSName),
-			CanonicalZoneID: aws.StringValue(loadbalancer.CanonicalHostedZoneNameID),
+	for {
+		resp, err := client.DescribeLoadBalancers(params)
+		if err != nil {
+			return nil, err
+		}
+		if resp == nil {
+			return nil, ErrInvalidAWSResponse
 		}
+
+		loadBalancers := resp.LoadBalancerDescriptions
+		log.Debugf("Page of elbs. Size: %d. Next marker: %v", len(loadBalancers), aws.StringValue(resp.NextMarker))
+		for _, loadbalancer := range loadBalancers {
+			result = append(result, &LoadBalancer{
+				DNSName:         aws.StringValue(loadbalancer.DNSName),
+				CanonicalZoneID: aws.StringValue(loadbalancer.CanonicalHostedZoneNameID),
+			})
+		}
+
+		if aws.StringValue(resp.NextMarker) == "" {
```

Comment:
this pagination helper could get rid of the manual marker stuff https://godoc.org/github.com/aws/aws-sdk-go/service/elb#ELB.DescribeLoadBalancersPages